modules/server/src/mapping/resolvers.ts

```diff
@@ -0,0 +1,114 @@
+			},
+			context: Context,
+			info: GraphQLResolveInfo,
+		) => Record<string, Aggregation>,
+		dataMaskThreshold: number,
+	) =>
+	async (obj: Root, args: Hits, context: Context, info: GraphQLResolveInfo) => {
+		/*
+		 * Get "aggregations" field from full query if found
+		 * Popular gql parsing libs parse the "info" property which may not include full query based on schema
+		 */
+		const aggregationsPath = 'operation.selectionSet.selections[0].selectionSet.selections';
+		const aggregationsSelectionSet = get(info, aggregationsPath, []).find(
+			(selection) => selection.kind === 'Field' && selection.name.value === 'aggregations',
+		);
+
+		/*
+		 * This function is used for "aggregation only mode" of Arranger where "hits" is based on "aggregations"
+		 * A user might request only the "hits" field in a GQL query, in which case return 0
+		 */
+		if (aggregationsSelectionSet) {
+			const modifiedInfo = { ...info, fieldNodes: [aggregationsSelectionSet] };
+			// @ts-ignore
+			// modifying the query info field inline so it can query aggregations correctly
+			// not idiomatic so doesn't line up with typings from graphql
+			const aggregations = await aggregationsQuery(obj, info.variableValues, context, modifiedInfo);
+			const { hitsTotal: total } = applyAggregationMasking({
+				aggregations,
+				thresholdMin: dataMaskThreshold,
+			});
+			return { total };
+		} else {
+			return { total: 0 };
+		}
+	};
+
+export const createResolvers = ({
+	createStateResolvers,
+	type,
+	Parallel,
+	getServerSideFilter,
+	dataMaskThreshold,
+	enableDocumentHits,
+}: Omit<CreateConnectionResolversArgs, 'enableAdmin'>) => {
+	// configs
+	const configs = async (parentObj: Root, { fieldNames }: { fieldNames: string[] }) => {
+		return {
+			downloads: type.config?.[ConfigProperties.DOWNLOADS],
+			extended: fieldNames
+				? type.extendedFields.filter((extendedField: ExtendedConfigsInterface) =>
+						fieldNames.includes(extendedField.fieldName),
+				  )
+				: type.extendedFields,
+			...(createStateResolvers && {
+				facets: type.config?.[ConfigProperties.FACETS],
+				matchbox: type.config?.[ConfigProperties.MATCHBOX],
+				table: type.config?.[ConfigProperties.TABLE],
+			}),
+		};
+	};
+
+	// aggregations
+	const aggregationsQuery = resolveAggregations({ type, getServerSideFilter });
+
+	const aggregations = async (
+		obj: Root,
+		args: {
+			filters: object;
+			include_missing: boolean;
+			aggregations_filter_themselves: boolean;
+		},
+		context: Context,
+		info: GraphQLResolveInfo,
+	) => {
+		const aggs = await aggregationsQuery(obj, args, context, info);
+		return aggregationsToGraphql(aggs);
+	};
+
+	// hits
+	const defaultHitsResolver = resolveHits({ type, Parallel, getServerSideFilter });
+	const hits = enableDocumentHits
+		? defaultHitsResolver
+		: // @ts-ignore
+		  // typing resolveAggregations requires typing a lot of code down the chain
+		  // TODO: improve typing
+		  resolveHitsFromAggs(aggregationsQuery, dataMaskThreshold);
```

Comment:
this is a point where we hit a wall with TS versus legacy JS. There's a lot of typing to be done because it starts to cascade into lots of JS functions.